sql/src/main/java/io/crate/execution/engine/aggregation/impl/ArrayAgg.java

```diff
@@ -42,19 +40,24 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import static io.crate.metadata.functions.TypeVariableConstraint.typeVariable;
+import static io.crate.types.TypeSignature.parseTypeSignature;
+
 public final class ArrayAgg extends AggregationFunction<List<Object>, List<Object>> {
 
     public static final String NAME = "array_agg";
 
     public static void register(AggregationImplModule module) {
-        module.register(NAME, new BaseFunctionResolver(FuncParams.SINGLE_ANY) {
-
-            @Override
-            public FunctionImplementation getForTypes(List<DataType> types) throws IllegalArgumentException {
-                var type = types.get(0);
-                return new ArrayAgg(type);
-            }
-        });
+        module.register(
+            Signature.builder()
+                .name(NAME)
+                .kind(FunctionInfo.Type.AGGREGATE)
+                .argumentTypes(parseTypeSignature("E"))
+                .typeVariableConstraints(typeVariable("E"))
+                .returnType(parseTypeSignature("array(E)"))
+                .build(),
```

Comment:
Maybe worth creating builder shortcuts like `Signature.scalar()` also for aggregates? I think this really improves readability. Also note the `signature.withTypeVariableContraints()` builder.